fastify.d.ts

```diff
@@ -0,0 +1,238 @@
+
+/// <reference types="node" />
+/// <reference types="pino" />
+
+import * as http from 'http';
+import * as pino from 'pino';
+
+declare function fastify(opts?: fastify.ServerOptions): fastify.FastifyInstance;
+
+declare namespace fastify {
+
+  type Plugin = (instance: FastifyInstance, opts: Object, callback?: (err?: Error) => void) => void
+
+  type Middleware = (req: fastify.FastifyRequest, res: fastify.FastifyReply, callback?: (err?: Error) => void) => void
+
+  type HTTPMethod = 'DELETE' | 'GET' | 'HEAD' | 'PATCH' | 'POST' | 'PUT' | 'OPTIONS';
+  
+  type BeforeRequestHandler = (req: FastifyRequest, reply: FastifyReply, done: (err?: Error) => void) => void
+  
+  type RequestHandler = (req: FastifyRequest, res: FastifyReply) => void
+
+  /**
+   * fastify's wrapped version of node.js IncomingMessage
+   */
+  interface FastifyRequest {
+    query: object,
+    body: any,
+    params: object,
+
+    req: http.IncomingMessage
+    log: pino.Logger
+  }
+
+  /**
+   * Response object that is used to build and send a http response
+   */
+  interface FastifyReply {
+    code: (statusCode: number) => FastifyReply
+    header: (name: string, value: any) => FastifyReply
+    type: (contentType: string) => FastifyReply
+    redirect: (statusCode: number, url: string) => FastifyReply
+    serializer: (fn: Function) => FastifyReply
+    send: (payload?: string|Array<any>|Object|Error|Promise<any>|ReadableStream) => FastifyReply 
+    sent: boolean
+  }
+
+  interface ServerOptions {
+    logger?: pino.LoggerOptions,
+    https?: boolean
+  }
+
+  interface JSONSchema {
+    // TODO - define/import JSONSchema types
+    body?: Object
+    querystring?: Object
+    params?: Object
+    response?: {
+      [code: number]: Object
+    }
+  }
+
+  /**
+   * Optional configuration parameters for the route being created
+   */
+  interface RouteShorthandOptions {
+    schema?: JSONSchema
+    beforeHandler?: BeforeRequestHandler
+  }
+
+  /**
+   * Route configuration options such as "url" and "method"
+   */
+  interface RouteOptions extends RouteShorthandOptions {
```

Comment:
Here there is not the full route declaration, checkout the [documentation](https://github.com/fastify/fastify/blob/master/docs/Routes.md#full-declaration)!

Reply:
Except `path` alias for `url`, I fail to see what is missing 🤔

Reply:
Ops, sorry! I missed the `extends` :P

Reply:
ah ah you're welcome ! Thx so much for the review & explanations !!!